CompanyApi/Controllers/CompanyController.cs

```diff
@@ -22,8 +23,73 @@ public ActionResult<Company> Create(CreateCompanyRequest request)
 
         [HttpDelete]
         public void ClearData()
-        { 
+        {
             companies.Clear();
         }
+
+        [HttpGet("{id}")]
+        public ActionResult<Company> Get(string id)
+        {
+            foreach (var company in companies.Where(company => company.Id.Equals(id)))
+            {
+                return Ok(company);
+            }
+
+            return NotFound();
+        }
+
+        [HttpGet]
+        public ActionResult<List<Company>> GetInPage([FromQuery] int? pageSize, [FromQuery] int? pageIndex)
+        {
+            if (pageIndex == null||pageSize == null)
+            {
+                return Ok(companies);
+            }
+            List<Company> companiesInPage = companies.Skip(((int)pageIndex -1)* (int)pageSize).Take((int)pageSize).ToList();

```

Comment:
well: nice to use Skip Take
suggestion: extract (pageIndex - 1) * pageSize to a temp variable